refund.sol

```diff
@@ -0,0 +1,91 @@
+/*
+
+CODE TO MAKE THE DAO INTO AN AUTOMATIC REFUND
+
+*/
+// A generic token contract
+contract token { 
+    mapping (address => uint256) public balanceOf;  
+    function transferFrom(address _from, address _to, uint256 _value) returns (bool success);
+    function mintToken(address target, uint256 mintedAmount);
+    uint256 public totalSupply;
+}
+
+
+/* define 'owned' */
+contract owned {
+    address public owner;
+
+    function owned() {
+        owner = msg.sender;
+    }
+
+    modifier onlyOwner {
+        if (msg.sender != owner) throw;
+        _
+    }
+
+    function transferOwnership(address newOwner) onlyOwner {
+        owner = newOwner;
+    }
+}
+
+contract DAORefund is owned {
+    // address of original DAO
+    token DAOTokens;
+    
+    // Address of new refund token (can be used to claim any additional eventual refund in the future)
+    token DAOBadgeOfHonor;
+    
+    // the price it will refund it for
+    uint public weiRefundedPerToken;
+
+    /* First time setup */
```

Comment:
Please provide some info here in the comments regarding the arguments. Especially some extra info aout the  BOHAddress which I assume is the 'Badge Of Honor' token address.